content/300-guides/300-upgrade-guides/200-upgrading-versions/600-upgrading-to-prisma-5.mdx

```diff
@@ -0,0 +1,677 @@
+prisma.post.create({
+  data: {
+    tags: 'databases',
+  },
+})
+
+prisma.post.findMany({
+  where: {
+    tags: 'databases',
+  },
+})
+```
+
+Prisma 5:
+
+```js highlight=3,9;normal
+prisma.post.create({
+  data: {
+    tags: ['databases'],
+  },
+})
+
+prisma.post.findMany({
+  where: {
+    tags: ['databases'],
+  },
+})
+```
+
+#### Composite lists
+
+Operations on lists of [Composite](/concepts/components/prisma-schema/data-model#defining-composite-types) types now only accept array values.
+
+<Admonition type="info">
+
+Composite lists are only available in the [MongoDB provider](/concepts/database-connectors/mongodb)
+
+</Admonition>
+
+With the following schema:
+
+```prisma
+model Post {
+  id           String    @id @default(uuid())
+  commentsList Comment[]
+}
+
+type Comment {
+  text String
+}
+```
+
+Prisma 4 and lower:
+
+```js
+prisma.post.findMany({
+  where: {
+    commentsList: {
+      equals: { text: 'hello' },
+    },
+  },
+})
+```
+
+Prisma 5:
+
+```js highlight=4;normal
+prisma.post.findMany({
+  where: {
+    commentsList: {
+      equals: [{ text: 'hello' }],
+    },
+  },
+})
+```
+
+<details><summary>Shorthand notation usage</summary>
+
+If you use the shorthand notation and exclude `equals`, you still must supply an array value for composite list fields.
+
+Prisma 4 and lower:
+
+```js
+prisma.post.create({
+  data: {
+    commentsList: { text: 'hello' },
+})
+
+prisma.post.findMany({
+  where: {
+    commentsList: { text: 'hello' },
+  },
+})
+```
+
+Prisma 5:
+
+```js highlight=3,8;normal
+prisma.post.create({
+  data: {
+    commentsList: [{ text: 'hello' }],
+})
+
+prisma.post.findMany({
+  where: {
+    commentsList: [{ text: 'hello' }],
+  },
+})
+```
+
+</details>
+
+### <inlinecode>cockroachdb</inlinecode> provider is now required when connecting to a CockroachDB database
+
+With Prisma version 5.0.0, we are requiring the CockroachDB instances be connected to with a `cockroachdb` provider. Previously, we had accepted `postgresql` as well, but we are removing that option.
+
+If you were using [native database types](/reference/api-reference/prisma-schema-reference#postgresql) and also the `postgresql` provider, you will most likely need to `diff` your database to see if anything has changed:
+
+1. Update your provider from `postgresql` to `cockroachdb`
+2. Backup your existing `prisma.schema` file
+3. Use `npx prisma db pull --force` in order to overwrite your existing schema (including native types) to those that are on your CockroachDB instance.
+4. Review changes between your schema backup and the new schema generated by `db pull`. You can either use the new schema as is, or update it to include your preferred spacing, comments, etc.
+5. delete your existing migrations. We will be [performing a baseline](/getting-started/setup-prisma/add-to-existing-project/relational-databases/baseline-your-database-typescript-cockroachdb) in order to make your local setup agree with your existing CockroachDB instance.
```

Comment:
So we are recommending baselining after all? No objections, I think that's probably on the safer side than recommending doing weird stuff with the migration lock file, I'm just a little confused at this point about what exactly was the result of the discussion.

Reply:
Yes. This is a pretty small subset of users, so I'm ok with giving them a safe, repeatable, and (perhaps most importantly) reversible process.